compiler/pipes/deduce-implicit-types-and-casts.cpp

```diff
@@ -807,7 +807,7 @@ void DeduceImplicitTypesAndCastsPass::on_clone(VertexAdaptor<op_clone> v_clone)
   kphp_error_return(v_clone->class_id, "`clone` keyword can be used only with instances");
   kphp_error(!v_clone->class_id->is_lambda_class(), "It's forbidden to `clone` lambdas");
   kphp_error(!v_clone->class_id->is_typed_callable_interface(), "It's forbidden to `clone` lambdas");
-  kphp_error(!v_clone->class_id->is_builtin() || v_clone->class_id->is_ffi_cdata(), "It's forbidden to `clone` built-in classes");
+  kphp_error(!v_clone->class_id->is_builtin() || v_clone->class_id->need_generated_stub || v_clone->class_id->is_ffi_cdata(), "It's forbidden to `clone` built-in classes");
```

Comment:
As far as I understand stubs are generated for builtin classes only. If so, I don't understand why do we need an additional check here.

Reply:
Yes, you are right that stub exists only for builtins. This condition check is class `not builtin` or `is_ffi_data`. Here also should be added classes, which are builtins, but need stubs